### PR TITLE
Migrate to tun2

### PR DESCRIPTION
**Pull request by ssrlive**
Since I have published tun2, so the `wintun.rs` can be deleted.

PS. Hi @SajjadPourali ,  I hope you can be a co-contributor and publisher of [tun2](https://crates.io/crates/tun2). If you wish I'm happy to sending inviting.

### CODE REVIEW

**Comment by ssrlive** [comment]
Welcome to commit a PR. @planetoryd

File: Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ tokio = { version = "1.34", features = [
 ], default-features = false }
 etherparse = { version = "0.13", default-features = false }
 thiserror = { version = "1.0", default-features = false }
-tracing = { version = "0.1.40", default-features = false }
```
Comment (planetoryd) [inline Cargo.toml:22]: I'm not sure this is a good choice. I ended up using both in nsproxy. I haven't done enough research on the two although its probably trivial

**Comment by planetoryd** [comment]
That was a quick n dirty fix. The problem is that I need backtrace. It is possible with `thiserror` itself right ?

**Comment by ssrlive** [comment]
yeah @planetoryd

File: Cargo.toml (hunk 1 above)
Comment (ssrlive) [reply Cargo.toml:22]: The modifications here are to increase the flexibility of using crates

**Comment by SajjadPourali** [comment]
> PS. Hi @SajjadPourali , I hope you can be a co-contributor and publisher of [tun2](https://crates.io/crates/tun2). If you wish I'm happy to sending inviting.

I'm certainly interested!
Thank you.

**Comment by SajjadPourali** [comment]
@ssrlive : Thanks! It looks good to me; however, I have some concerns about FreeBSD support since NarrowLink is officially available in the [FreeBSD ports](https://www.freshports.org/net/narrowlink/).

File: src/lib.rs [hunk 2]
```diff
@@ -31,7 +32,7 @@ const TTL: u8 = 64;
 #[cfg(target_os = "windows")]
 const TTL: u8 = 128;
 
-#[cfg(not(target_os = "windows"))]
+#[cfg(any(target_os = "linux", target_os = "macos"))]
```
[the remark this answers is not in the document]
Comment (ssrlive) [reply src/lib.rs:35]: How about changing it to `#[cfg(any(target_os = "linux", target_os = "freebsd", target_os = "macos"))]`?

[the remark this answers is not in the document]
Comment (planetoryd) [reply src/lib.rs]: @ssrlive Now you are ignoring OpenBSD.

[the remark this answers is not in the document]
Comment (ssrlive) [reply src/lib.rs]: now it's become `#[cfg(unix)]`

**Pull request merged**